openquake/server/views.py

```diff
@@ -700,6 +701,34 @@ def web_engine_get_outputs(request, calc_id, **kwargs):
                   dict([('calc_id', calc_id)]))
 
 
+@csrf_exempt
+@cross_domain_ajax
+@require_http_methods(['POST'])
+def on_same_fs(request):
+    """
+    Accept a POST request to check access to a FS available by a client.
+
+    :param request:
+        `django.http.HttpRequest` object, containing mandatory parameters
+        filename and checksum.
+    """
+    filename = request.POST['filename']
+    checksum_in = request.POST['checksum']
+
+    checksum = 0
+    try:
+        data = open(filename, 'rb').read(32)
+        checksum = zlib.adler32(data, checksum) & 0xffffffff
+        if checksum == int(checksum_in):
+            return HttpResponse(content=json.dumps({'success': True}),
+                                content_type=JSON, status=200)
+    except:
+        pass
```

Comment:
I guess the expected error here is an IOError when the user passes a file path which does not exist. So I would trap only IOError.